### PR TITLE
Fix floating point comparison in JTC

### PR DESCRIPTION
**Pull request by Abishalini**
I was going through the MoveIt tutorials and there were instances where [motion planning succeeded but execution failed](https://github.com/ros-planning/moveit2_tutorials/issues/828) with the error  
```
[ros2_control_node-5] [ERROR] [1701709582.650235815] [joint_trajectory_controller]: Velocity of last trajectory point of joint joint_1 is not zero: -0.000000
```
Changing the comparison to use an epsilon fixed it locally for me.

### CODE REVIEW

**Review by destogl — APPROVED** [review]

File: joint_trajectory_controller/src/joint_trajectory_controller.cpp [hunk 1]
```diff
@@ -1369,10 +1369,10 @@ bool JointTrajectoryController::validate_trajectory_msg(
   {
     for (size_t i = 0; i < trajectory.points.back().velocities.size(); ++i)
     {
-      if (trajectory.points.back().velocities.at(i) != 0.)
+      if (fabs(trajectory.points.back().velocities.at(i)) > std::numeric_limits<float>::epsilon())
       {
         RCLCPP_ERROR(
-          get_node()->get_logger(), "Velocity of last trajectory point of joint %s is not zero: %f",
+          get_node()->get_logger(), "Velocity of last trajectory point of joint %s is not zero: %.15f",
           trajectory.joint_names.at(i).c_str(), trajectory.points.back().velocities.at(i));
```
Comment (christophfroehlich) [inline joint_trajectory_controller/src/joint_trajectory_controller.cpp:1376]: our CI complains about the format (see our contribution guidelines on how to install pre-commit)
```suggestion
          get_node()->get_logger(),
          "Velocity of last trajectory point of joint %s is not zero: %.15f",
          trajectory.joint_names.at(i).c_str(), trajectory.points.back().velocities.at(i));
```

**Review by christophfroehlich — APPROVED** [review]
LGTM, only one small fix in the format.

**Review by saikishor — APPROVED** [review]
LGTM

**Comment by Abishalini** [comment]
Can this be backported to humble as well?

**Pull request merged**